src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -184,13 +197,17 @@ def run(
                                 "Current thread is not main thread, skip signal handler registration in BatchEngine."
                             )
 
+                    if self._chat_group_roles is None:
```

Comment:
if you have already used `ExecutorProxy` to hide the details of chat group, is it still a good idea to make `BatchEngine` aware of `chat_group_roles`?

@PeiwenGaoMS @thy09  to comment on this

Reply:
I guess here what we actually need is a parameter like `self._should_apply_inputs_mapping`?